browser/ui/webui/settings/brave_social_blocking_handler.cc

```diff
@@ -0,0 +1,50 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/webui/settings/brave_social_blocking_handler.h"
+
+#include <string>
+
+#include "base/bind.h"
+#include "base/values.h"
+#include "brave/common/pref_names.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/common/pref_names.h"
+#include "components/prefs/pref_service.h"
+#include "content/public/browser/web_ui.h"
+#include "content/public/browser/web_ui_data_source.h"
+
+BraveSocialBlockingHandler::BraveSocialBlockingHandler()
+    : weak_ptr_factory_(this) {}
+
+BraveSocialBlockingHandler::~BraveSocialBlockingHandler() {}
+
+void BraveSocialBlockingHandler::RegisterMessages() {
+  profile_ = Profile::FromWebUI(web_ui());
+  web_ui()->RegisterMessageCallback(
+      "setGoogleLoginEnabled",
+      base::BindRepeating(&BraveSocialBlockingHandler::SetGoogleLoginEnabled,
+                          base::Unretained(this)));
+}
+
+void BraveSocialBlockingHandler::SetGoogleLoginEnabled(
+    const base::ListValue* args) {
+  CHECK_EQ(args->GetSize(), 1U);
+  CHECK(profile_);
+  bool enabled;
+  args->GetBoolean(0, &enabled);
+
+  profile_->GetPrefs()->SetBoolean(prefs::kSigninAllowedOnNextStartup, enabled);
+}
+
+// static
+void BraveSocialBlockingHandler::AddLoadTimeData(
+    content::WebUIDataSource* data_source,
+    Profile* profile) {
+  data_source->AddBoolean(
+      "googleLoginEnabledAtStartup",
```

Comment:
it's confusing to me that we're using a different name for this pref in JS and also why are we aren't we just using the prefs JS api for this? cc @petemill

Reply:
I don't know if this is the right pref to use anyway from https://github.com/brave/brave-core/pull/5616/files#r434915558 but using a different name in JS and cpp confuses things. We should use consistent naming internally

Reply:
@bridiver - This code is used to figure out when to show the restart button when the setting is toggled. `AddLoadTimeData` sets the initial value on start of `kSigninAllowedOnNextStartup`. This is compared with the current value to figure out if the `restart button` should/should-not be displayed.

Reply:
What I'm saying is that I think changing the naming is confusing. It's much better imo to use consistent naming so it's easy to navigate through the code and not have to try to figure out what `googleLoginEnabledAtStartup` maps to. Also I believe we could skip this handler entirely because you can handle prefs directly in JS